base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs

```diff
@@ -562,7 +568,9 @@ where TBackend: TransactionBackend + 'static
             .await?
             .values()
             .filter(|tx| {
-                tx.status == TransactionStatus::MinedUnconfirmed || tx.status == TransactionStatus::MinedConfirmed
+                tx.status == TransactionStatus::Broadcast ||
```

Comment:
Why did you add the Broadcast here? I don't think you can validate a Broadcast transaction because it is not in the chain? If it is in the Broadcast state it will be checked when a broadcast protocol is started for it.

Reply:
This is to enable the wallet to update its transaction status after a reorg. It may have broadcasted a transaction using a UTXO that has been reorged out, same as the other two included there. In that case the broadcast monitoring protocol must be started so the transaction can be invalidated.

Reply:
But this will invalidate normal transactions that have been broadcast because they are not in the chain yet. You cannot validate a transaction that is not mined.